### PR TITLE
Bug fix for out of Hints but alternate expectation fulfilled

### PR DESCRIPTION
**Pull request by dan100110**

### CODE REVIEW

File: node/test/dialog.spec.ts [hunk 1]
```diff
@@ -150,6 +194,8 @@ describe('dialog', async () => {
             return [200, { data: { lesson: navyIntegrityLesson } }];
           } else if ((reqBody.query as string).includes('q2')) {
             return [200, { data: { lesson: currentFlowLesson } }];
+          } else if ((reqBody.query as string).includes('q3')) {
```
Comment (beatthat) [inline node/test/dialog.spec.ts:197]: this hack for mapping the query from gql to Lesson data object has outlived its welcome.

We don't need a super elegant fix right now, but we should make it less egregious.

I recommend the following solution:

```typescript
const lessonById: Record<string, Lesson> = {
  q1: {...},
  q2: {...},
  q3: {...},
}

function findLessonForGqlQuery(query: string) : Lesson {
  lessonId = (query.includes("q1"))
    ? "q1"
    : (query.includes("q2")
    ? "q2"
    : (query.includes("q3")
    ? "q3"
    : "ok fix this properly with a regex already";
  return lessonById[lessonId]
}
```

**Review by beatthat — CHANGES_REQUESTED** [review]
looks good! just a couple of house-keeping changes requested (see inline)

**Review by beatthat — CHANGES_REQUESTED** [review]
one more tiny change requested about how to test for empty and then looks good to go

**Review by beatthat — APPROVED** [review]
looks good!
